File: client-domain/src/test/java/com/jraska/github/client/DeepLinkNavigatorTest.java
package com.jraska.github.client;

import org.junit.Test;

import okhttp3.HttpUrl;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class DeepLinkNavigatorTest {
  @Test
  public void whenStartUserDetail_thenCorrectDeepLinkLaunched() {
    DeepLinkLauncher deepLinkLauncher = mock(DeepLinkLauncher.class);
    WebLinkLauncher webLinkLauncher = mock(WebLinkLauncher.class);
    DeepLinkNavigator navigator = new DeepLinkNavigator(deepLinkLauncher, webLinkLauncher);

    navigator.startUserDetail("johny");

    verify(deepLinkLauncher).launch(HttpUrl.parse("https://github.com/johny"));
  }
}

File: core/src/test/java/com/jraska/github/client/DeepLinkNavigatorTest.kt
package com.jraska.github.client

import okhttp3.HttpUrl.Companion.toHttpUrl
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class DeepLinkNavigatorTest {
  @Test
  fun whenStartUserDetail_thenCorrectDeepLinkLaunched() {
    val deepLinkLauncher = mock(DeepLinkLauncher::class.java)
    val webLinkLauncher = mock(WebLinkLauncher::class.java)
    val navigator = DeepLinkNavigator(deepLinkLauncher, webLinkLauncher)

    navigator.startUserDetail("johny")

    verify(deepLinkLauncher).launch("https://github.com/johny".toHttpUrl())
  }
}
